### PR TITLE
SLVS-1401 Add new BoundProject model

### PR DESCRIPTION
**Pull request by georgii-borovinskikh-sonarsource**

### CODE REVIEW

File: src/Core/Binding/BoundServerProject.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+namespace SonarLint.VisualStudio.Core.Binding;
+
+public class BoundServerProject
+{
+    public string ProjectKey { get; }
```
Comment (gabriela-trutan-sonarsource) [inline src/Core/Binding/BoundServerProject.cs:25]: Is this the projectKey as in the VisualStudio solution name or the projectKey from the server? 
Should we also not add the solution name and its type, e.g., see ConnectedModeProject for reference?

File: src/Core/Binding/BoundServerProject.cs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+namespace SonarLint.VisualStudio.Core.Binding;
+
+public class BoundServerProject
```
Comment (gabriela-trutan-sonarsource) [inline src/Core/Binding/BoundServerProject.cs:23]: This model will not be used for serialization?

**Review by gabriela-trutan-sonarsource — CHANGES_REQUESTED** [review]

File: src/Core/Binding/BoundServerProject.cs (hunk 2 above)
Comment (georgii-borovinskikh-sonarsource) [reply src/Core/Binding/BoundServerProject.cs:23]: Yes

File: src/Core/Binding/BoundServerProject.cs (hunk 1 above)
Comment (georgii-borovinskikh-sonarsource) [reply src/Core/Binding/BoundServerProject.cs:25]: We don't have that in the old binding model, but it's a good point to add it. I will do that.

**Review by gabriela-trutan-sonarsource — APPROVED** [review]

**Pull request merged**